ITE/3/README.adoc

```diff
@@ -0,0 +1,596 @@
+= ITE-3: Real-world example of combining TUF and in-toto for packaging Datadog Agent integrations
+:source-highlighter: pygments
+:toc: preamble
+:toclevels: 5
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+.Metadata
+[cols="2"]
+|===
+| ITE
+| 3
+
+| Title
+| in-toto Enhancement format
+
+| Sponsor
+| link:https://github.com/trishankatdatadog[Trishank Karthik Kuppusamy]
+
+| Status
+| Draft :pensive:
+
+| Type
+| Informational
+
+| Created
+| 2019-07-19
+
+|===
+
+[[abstract]]
+== Abstract
+
+Datadog is a monitoring service for cloud-scale applications that monitors
+servers, databases, tools, and services through a
+software-as-a-service-based data analytics platform. It supports
+multiple cloud service providers, including Amazon Web Services,
+Microsoft Azure, Google Cloud Platform, and Red Hat OpenShift. At the
+time of this writing, the company servers more than 8,000 customers, and
+collects trillions of monitoring record points on a daily basis.
+
+The Datadog https://docs.datadoghq.com/agent/?tab=agentv6[agent] is the
+software that runs on virtual machines or containers. It collects events and
+metrics from these virtual machines or containers and sends them to Datadog,
+where customers can analyze their monitoring and performance data. The agent
+https://docs.datadoghq.com/getting_started/integrations/[integrations]
+are plug-ins that collect metrics from services running on customer
+infrastructure. Presently, there are more than one hundred integrations
+that come installed out-of-the-box with the Agent.
+
+This ITE discusses the TUF security model used to distribute the Datadog Agent
+integrations in a compromise-resilient manner.
+
+[[motivation-and-reasoning]]
+== Motivation and reasoning
+
+See
+xref:../2/README.adoc#motivation-and-reasoning[ITE-2].
+
+[[specification]]
+== Specification
+
+[[datadog-agent-integrations]]
+=== Datadog Agent integrations
+
+image:tuf-in-toto-figure-1.png[Figure 1]
+
+*Figure 1*: The Datadog Agent integrations software supply chain.
+
+The Datadog agent integrations supply chain, shown in Figure 1,
+presently has three steps:
+
+[arabic]
+. The first "`tag`" step outputs Python source code as products. Every
+integration consists of Python source code and several
+https://yaml.org/[YAML] configuration files. The step uses (1, n)
+Yubikeys, where n is the current number of Datadog Agent integration
+developers. (See this
+https://dd-integrations-core-wheels-build-stable.datadoghq.com/targets/in-toto-metadata/3e592d47ab815cb9bd8aa064bbaa2a217c1baa9a55a7f8b73d2da6dfd14a247e/tag.c295cf63.link[example]
+"`tag`" link metadata file.)
```

Comment:
Can you give a little context of why you made these choices?  What you did is valuable to know.  Why you did it is even better!